app/Http/Controllers/RegisterController.php

```diff
@@ -1,19 +1,136 @@
-<?php
+<?php declare(strict_types=1);
 
 namespace App\Http\Controllers;
 
-use Illuminate\Http\{
-    Request,
-    JsonResponse
+use Exception;
+use Illuminate\Http\Request;
+use Illuminate\Support\Arr;
+
+use App\Models\{
+    Organization,
+    OrganizationSetting,
+    User
+};
+
+use Illuminate\Support\Facades\{
+    DB,
+    Hash,
+    Validator
 };
 
-use Inertia\Inertia;
-use Inertia\Response as InertiaResponse;
+use Inertia\{
+    Inertia,
+    Response as InertiaResponse
+};
 
-class RegisterController extends Controller
+class RegisterController extends AppController
 {
-    public function showRegister(Request $request): InertiaResponse
+    /**
+     * render the registration page
+     *
+     * @return InertiaResponse
+     */
+    public function showRegister(): InertiaResponse
     {
         return Inertia::render('Register');
     }
+
+    /**
+     * process new user registration
+     *
+     * @param Request $request
+     * @return InertiaResponse
+     */
+    public function register(Request $request): InertiaResponse
+    {
+        try {
+            $payload = $this->getValidatedRegistrationPayload($request->json()->all());
+            $this->submitRegistrationPayload($payload);
+        } catch (Exception $e) {
+            return Inertia::render('Register', [
+                'error' => $e->getMessage()
+            ]);
+        }
+
+        return Inertia::render('Register', [
+            'success' => 'New account created successfully.'
+        ]);
+    }
+
+    /**
+     * submit new user credentials
+     *
+     * @param array $payload
+     * @return void
+     */
+    private function submitRegistrationPayload(array $payload): void
+    {
+        DB::transaction(function () use ($payload): void {
+            $user = User::create([
+                'name'     => $payload['firstname'] . ' ' . $payload['lastname'],
+                'email'    => $payload['email'],
+                'password' => Hash::make($payload['password'])
+            ]);
+
+            if (isset($payload['organization'])) {
+                $orgSetting = OrganizationSetting::create([
+                    'point_of_contact_id' => $user->id
+                ]);
+
+                $org = Organization::create([
+                    'org_name'                => $payload['organization'],
+                    'organization_setting_id' => $orgSetting->id,
+                    'active'                  => 1
+                ]);
+
+                User::where('id', $user->id)
+                    ->update([
+                        'organization_id' => $org->id
+                    ]);
```

Comment:
Can you use `(new Organization)->createNewOrg($payload['organization'], $user->id);` instead?

Reply:
This is an example where we now have the create org in two controllers. The OrgController can be used for an admin purpose and this registration controller has its own purpose. But both purposes need to create an org the same way. By putting it in the model now it will be easier for us to add events to it and get the same behaviour everywhere that orgs can be created.